src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs

```diff
@@ -311,39 +331,45 @@ private static void WriteRequestBuilderConstructorCall(CodeMethod codeElement, L
                 writer.WriteLine(conventions.DocCommentEnd);
             }
         }
-        private string GetDeserializationMethodName(CodeTypeBase propType) {
+        private string GetDeserializationMethodName(CodeTypeBase propType, CodeMethod method) {
             var isCollection = propType.CollectionKind != CodeTypeBase.CodeTypeCollectionKind.None;
-            var propertyType = conventions.TranslateType(propType);
+            var propertyType = conventions.GetTypeString(propType, method, false);
             if(propType is CodeType currentType) {
                 if(isCollection)
                     if(currentType.TypeDefinition == null)
                         return $"n.getCollectionOfPrimitiveValues({propertyType.ToFirstCharacterUpperCase()}.class)";
+                    else if (currentType.TypeDefinition is CodeEnum enumType)
+                        return $"n.getCollectionOfEnumValues({enumType.Name.ToFirstCharacterUpperCase()}.class)";
                     else
                         return $"n.getCollectionOfObjectValues({propertyType.ToFirstCharacterUpperCase()}.class)";
                 else if (currentType.TypeDefinition is CodeEnum currentEnum)
                     return $"n.getEnum{(currentEnum.Flags ? "Set" : string.Empty)}Value({propertyType.ToFirstCharacterUpperCase()}.class)";
             }
             return propertyType switch
             {
-                "String" or "Boolean" or "Integer" or "Float" or "Long" or "Guid" or "OffsetDateTime" => $"n.get{propertyType.ToFirstCharacterUpperCase()}Value()",
+                "byte[]" => "n.getByteArrayValue()",
+                "String" or "Boolean" or "Integer" or "Float" or "Long" or "Guid" or "OffsetDateTime" or "Double" => $"n.get{propertyType.ToFirstCharacterUpperCase()}Value()",
```

Comment:
The property type already has an uppercaseFirst character. Thus, the call to `ToFirstCharacterUpperCase` is unnecessary. Yes?

Reply:
```suggestion
                "String" or "Boolean" or "Integer" or "Float" or "Long" or "Guid" or "OffsetDateTime" or "Double" => $"n.get{propertyType}Value()",
```